File: api/GameGuesserAPI/Models/Game.cs
using System.Globalization;

namespace GameGuesserAPI.Models
{
    public class Game
    {
        [BsonId] 
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; } 
        public string Name { get; set; }
        public string Genre { get; set; }
        public List<string> Platforms { get; set; }
        public int ReleaseYear { get; set; }
        public string Developer { get; set; }
        public string Publisher { get; set; }
        public string Description { get; set; }
        public string CoverImageUrl { get; set; }
        public Double Budget { get; set; }
        public string Saga { get; set; }
        public string POV { get; set; }
        public List<string> Keywords { get; set; }
        public List<string> Clues { get; set; }
    }
}

File: android-app/app/src/main/java/com/example/gameguesser/data/Game.kt
package com.example.gameguesser.data

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import com.google.gson.annotations.SerializedName

@Entity(tableName = "games")
@TypeConverters(GameConverters::class)
data class Game(

    @Ignore
    @SerializedName("_id")
    var _id: IdObject? = null,  // holds the Mongo _id

    @PrimaryKey
    var id: String = "",

    var name: String = "",
    var genre: String = "",
    var platforms: List<String> = emptyList(),
    var releaseYear: Int = 0,
    var developer: String = "",
    var publisher: String = "",
    var description: String = "",
    var coverImageUrl: String = "",
    var budget: Double = 0.0,
    var saga: String = "",
    var pov: String = "",
    var clues: List<String> = emptyList(),
    var keywords: List<String> = emptyList(),

    ) {

    val mongoId: String
        get() = _id?.oid ?: id

    constructor() : this(_id = null)
}

data class IdObject(
    @SerializedName("\$oid")
    val oid: String = ""
)
